solarkraft/src/verify.ts

```diff
@@ -15,6 +11,168 @@ import path from 'node:path'
 const APALACHE_DIST = '/opt/apalache'
 const APALACHE_BIN = path.join(APALACHE_DIST, 'bin', 'apalache-mc')
 
+// TODO(#38): Replace stubs with ITF
+type Value = { type: string; value: any }
+type Binding = { name: string; type: string; value: any }
+type State = Binding[]
+type Transaction = {
+    functionName: string
+    functionArgs: Value[]
+    env: { timestamp: number }
+    error: string
+}
+
+/**
+ * Return an instrumented monitor specification.
+ * @param monitor A TLA+ monitor, as returned from `apalache typecheck --output=temp.json`
+ * @param state The blockchain state before executing the transaction / the monitor.
+ * @param tx The transaction being applied to `state`.
+ * @returns The instrumented TLA+ monitor.
+ */
+export function instrumentMonitor(
+    monitor: any,
+    state: State,
+    tx: Transaction
+): any {
+    // Add a special variable `last_error` that tracks error messages of failed transactions
+    state.push({ name: 'last_error', type: 'TlaStr', value: '' })
+
+    // Declaration of  "Init" (according to `state`)
+    const tlaInit = tlaJsonOperDecl__Conjunction(
```

Comment:
```suggestion
    const tlaInit = tlaJsonOperDecl__And(
```

It would be just a shorter name

Reply:
Done 9448d16